### ISSUE TITLE
use case: forming projects, reconciling real/virtual accounts, auditing projects/accounts

### ISSUE THREAD

**Issue by ghost** [valueflows/exchange#15]
This is quite rough and ready whilst I have a few moments. The use case is based around a project I am involved in called [Robin Hood Coop](http://robinhoodcoop.org/).
### Background (Robin Hood Coop v.0.1)

We are a motley-crew actively looking for ways of funding and expanding the commons (what p2p foundation calls [commonfare](http://p2pfoundation.net/Commonfare)). This is the context for all our our experiments.

In 2012 we released a data mining algorithm which trades on Wall Street. We established a cooperative in Finland built around this data mining algorithm. This legal form allowed us to invest member payments on Wall Street through the mechanism of offering equity to our members (in Finland coops may do with their assets as they please, including investing on Wall Street, without requiring a Financial Services licence). Members have been required to pay a one time €30 membership fee and also one mandatory €30 share. There is no upper limit to how many voluntary shares a member may purchase beyond this.

When a member buys shares (equity), we ask them to choose a sharing scale for the profits. If a member decides to keep 100% of profits to themselves, then we have a one time tax of 10% of their total payment. This goes to Robin Hood Projects Fund. If a member chooses to share 50% of the profits with RHPF then they are not taxed on their payment. Shares are immutable - their sharing model cannot be changed once it is established, however, ownership of the share can change, or the share can be destroyed, once it is purchased back by RHC (when the member chooses this exit route).

Increases in value are only realised at Exit from the Coop.

There are 2 ways to exit the coop. 
- The coop can buy back the shares at the request of the member. This is a long process. To comply with Finnish law around cooperatives, we must wait till the end of the financial year and after our books have been audited by accountants (Ernst&Young). The funds are then moved to a holding account for 6 months before they can be passed to the member (I believe this is a rudimentary mechanism to stop theft of coop funds by quick movements)
- A member may purchase a share directly from another member and pass on the entitlement. 

---

The mandatory share gives the following rights:
- 1 vote (no matter how many voluntary shares a member might have - there is only one vote per member)
- Allowed to attend and vote in the AGM
- Allowed to propose projects which should be funded
- Allowed to propose themselves to be in the selection committee for projects.
### Project Selection
- RHPF is managed by three members who form a selection committee
- Selection committee is selected at random from the pool of members who have proposed themselves
- The committee must unanimously agree on which projects should be funded (they get to decide on all the terms)
- Projects selected must 'Expand the Commons'
- \- this is decided upon by the Board Members

---

This is how it has worked since 2012. In the next update I will talk about how the 'business operations' have been working to make the above system function

I will then follow this up with a post about the current transmutation we are going through. You can read about this [here](http://discourse.robinhoodcoop.org/t/moving-on-to-blockchain/259?u=danhassan)

This will be released within the next month.

---> it is this mutation that I would find very interesting to go through with you ValueFlows folks, as it collides very nicely with this use case: https://github.com/valueflows/exchange/issues/6

**Comment by fosterlynn** [comment]
@dan-mi-sun awesome to see your use case over here in VF!  I'm not sure how the blockchain mutation will impact the mapping, but I'd be interested in being part of working it through in gory detail whenever it makes sense.  And in conjunction with #6.

**Comment by ghost** [comment]
### ... Part Two

(sun is emerging from behind some clouds, still kinda overcast)

### Business Operations of Making Robin Hood Coop v.0.1 Work (Grind)

Ok - I'll try to map out what went into making this work. 

I'll preface this by saying Robin Hood Coop has been 95% run by volunteers. There are only two paid (ish) roles within the entire organisation. I'll point those out as we go along, to try and capture as much of the flows of $$$ as possible + the flows of work (blood, joy, sweat, tears)

**Warning: It's not always/ever very pretty, also - there may be some inaccuracies - but the general gist is there.**

**Also RHC website is currently down, until we roll the new system, this week or next**

---

### On Ramp:

*_note: the average number of emails a member sends throughout the signup process is nearly 4 emails. 4. Four. I won't highlight where these come up... but assume as you are reading the below, that it is a bit like wading through fudge *_

As long as there are no hiccups the on-ramping takes about 15 minutes per person. 1 -5 . Fifteen. _sigh_

#### New Member

Someone hears about RHC and decides to join. They arrive at the website and sign up with the following details:
- Name
- Address
- Email
- Password (they set this)

They are then passed through the the 'Buy Shares' section. This includes 
- Membership fee
- Mandatory Share
- They can then choose any number of voluntary shares
- They also choose the 'sharing' amount for this batch of shares.

This triggers an email in the system which is sent to the prospective member. This email contains instructions on how to send money via Direct Bank to RHC's bank in Finland.

The member is then responsible for sending money via direct bank transfer to RHC.

What this has meant in the past that the website holds 0 (zero) financial data. This has been great.

#### Robin Hood Coop Angel (Admin)

The admin is paid about €1000 - 2000 per month. For this they are the main person who persons the email and handles everything I am about to describe (or at least 80% of the time). They are the unsung heros - as nothing would work without them.

On the admin side RHC can see when someone new has signed up (an automated email is sent to the collective contact@ inbox)

Each day an Angel needs to go into the Finnish bank account and manually check new payments that have arrived into the account.
- If there is a new payment she then goes into the Admin section of RHC and looks for the corresponding payment with correct 'note' (did I mention the member needs to manually add this to payment?). If the member forgot this, then the Angel must don the Sherlock Holmes cap, insert a pipe into her mouth and start her detective work of figuring out who this payment is for.
- Once the payment can be attached to a user the RHC system must be updated manually (manually entering the number of shares, attaching the pdf of the bank payment receipt.
- Once this happens the Angel changes the status of the member from pending payment to paid (but not yet invested).
- The last thing the Angel does is to update an offline Spreadsheet which is meant to mimic the state of the DB... so she then fills in all the fields to make sure the double entry is happy.

**NOTE: there are often discrepancies, people send too much, too little, don't send the same $$$ as what they ordered etc etc etc**

---

#### Money, money, money

So, once the member payment comes into the RobinHoodCoop Finnish bank account this Money is abstractly marked between the following:
- assigned for the member account
- assigned for RHPF (either through voluntary sharing, or by a one off tax if sharing has not been chosen)
- if there is a membership fee this is all set aside for RHC

There are four accounts to keep in mind:

A. RHCs Finnish Bank Account (receives payments for new members)
B. RHCs Finnish 'holding account (for members funds who are 'cashing-out of the coop')
C. RHCs USA bank account
D. RHCs Interactive broker account

Each month money is kept for Operational Costs, the rest has been sent through to Interactive Brokers)... more on this later.

---

#### Back to the Member

With the change of their status, they receive an email and can now go and look at their shares and see the payment has been received but that the monies are not yet invested (more on this later).

_Aside_:
From an automation point of view this is all an Epic Sadness. Much manual input for all parties. Much sadness.

#### STATE (new member joined)

By this point a new member has signed up, they have sent money by direct bank transfer, an Angel has checked the bank account, has confirmed the payment, has assigned the payment to the correct member, has updated the members share payment and has changed their state.

For a first time member this is the point they are born as a denizen of Robin Hood. They are free to attend the AGM (which [we host in loomio](http://blog.loomio.org/2016/01/05/robinhood/)). They are able to propose projects and they are able to propose themselves for the Project Selection Committee.

### Each month (Admin)

At the beginning of each month all new member payments are bundelled together and transferred from the Finnish RHC bank account to our Interactive Broker account to be Invested on Wall Street.

Once this happens an Admin updates the system, and all members who joined in that period have their status updated and they can see a table of:

| Amount Invested | Date Paid | Amount of Personal Profit | Amount of Profit to RHPF

---

That's enough for this Chapter.  You've made it this far, here's a picture of a [cat playing with a computer](https://s-media-cache-ak0.pinimg.com/736x/7e/54/35/7e54353877944bb9c07c1b40c5171bf1.jpg)

---

Questions welcome at any point in this process... where I see inaccuracies, I will edit the post and correct.

Once I am finished with this process I will also likely get a few of my workmates to have a look over the information to double check the work.

**Comment by ghost** [comment]
@tereensio could you have a look at this and let me know if I have gotten this all right so far. I know your hands are deeper in the operational gears than mine :)

**Comment by ghost** [comment]
Here is a quick overview of the 'how we internally organise' at Robin Hood. Although a bit old, it still holds pretty true. It is more focusing on the communications and organisational side of things.

http://discourse.robinhoodcoop.org/t/a-tale-of-organising-robin-hood/117

**Comment by ghost** [comment]
### .... Part Three

(the sun has set, the sky has filled yet again with the pinpricked velvet sheet)

#### Irrigating the Commons, Robin Hood Coop v.0.1 Work (Grind): Robin Hood Projects Fund

What has been an interesting aspect of RHC is that it plays with value. It is not peoples charity, but a speculative contribution (i.e. people are speculating the algortihm in the market will lead to an increase in value, which can then be collectively managed). 

---

2012, 2013. 2014 all profits were reinvested back into the cooperative.

By 2015 the coop had accrued €150k so put a call out for projects.

Members (and friends of members) proposed nearly 100 projects. Each project was considered by the Board Members before they were made public. The projects were vetted with the following criteria: "Does this project expand the commons". Board Members are not paid for their work.

Project Submissions were made via the `contact@` email. A list of these projects went up on a wordpress site projects.robinhoodcoop.org (currently offline)

A group of about 15 people proposed themselves to be on the Selection Committee.

They proposed themselves by emailing the `contact@` email and Admin Angels made a list in a google doc.

15 names were each assigned a numbers. These numbers were assigned corks and then they were selected at random. You can see this [here](https://vimeo.com/125433301)

3 People were selected. You can see in their own words how they made the choice of three projects each receiving a grant. [Brian Massumi](https://vimeo.com/132813909) and [Catarina Saraiva](https://vimeo.com/135030069)

In a nutshell they deliberated long and hard. And chose based on extensive work and consideration. They were not paid for this work.

---

#### Money, Money, Money

Once allotted, the funds were withdrawn from account D to account C... and to be honest.. I am not sure how the money got to the end projects... but it did find its way to them

---

Rudimentary. But in a nutshell that is how the process 'worked'.

---

We are getting near to the point of speaking now about the new systems we have built (the transmutation) & then the proposed next developments...

Well done. You've made it to the end of another chapter. Here's a cool picture I found from Google Pics when typing the word ['transmutation'](http://media.gettyimages.com/photos/alchemical-symbol-of-transmutation-of-base-metal-into-gold-and-silver-picture-id113633503)

**Comment by ghost** [comment]
### ... Part Four

(the midnight hour)

### Robin Hood Coop v.1.0

This latest round of developments has been delivered by the Robin Hood Collective. This is a crew of people most of whome have been involved in Robin Hood Coop since the early days, but are also now involved in a special purpose entity (a Benefit Corporation incorporated in Delaware, NY, USA) called Robin Hood Services (RHS). 

Too much volunteerism == burn out.

RHS has been seeking $$$ from, somewhere - currently doing the rounds in Silicon Valley.  This is context for the transmutation. 

To really irrigate the commons, we need fertile financial ground. 

The Robin Hood Collective's next experiment is in delivering a commons financial product in which regular folk can come to 'do finance'. Our first customer is... ourselves; Robin Hood Coop. 

[Perhaps this is this](http://www.adinnerguest.com/wp-content/uploads/427110_513890661970095_1699119063_n.jpg).

That being said... a good place to start i in trying to solve your own problems.

#### Recap

The Robin Hood Collective has produced a new product. This production has been funded by assets from Robin Hood Coop, but the development of which has been Researched & Developed by the team at Robin Hood Services.

---

#### What can the new product do?

The new Robin Hood Coop product can do the following:

#### Share Management System
- Share representation as digital tokens on Ethereum blockchain
- RobinHood is able to issue an unlimited number of series of assets representing different share classes in RobinHood Cooperative
- RobinHood customers are able to purchase such shares from RobinHood. Share allocations are associated with private keys held on behalf of RobinHood’s customers
- RobinHood is able to make distributions to shareholders (i.e. dividend payments)
- users can buy tokens for fixed amount from issuer

##### Secondary Share Market
- users can transfer assets to other users as described in section 2. (“Payment Integrations”) below.
- RobinHood and RobinHood customers are able to transfer assets to any other user at any time (optionally subject to limitations set by RobinHood)
  ￼
  
  ##### User
- API and user interface components
- user registration
- user log­in
- private blockchain key management service
- user can download their own private key
- user can retrieve user’s own e­mail address and blockchain address
- support for industry­standard SSL encrypted communication
- optional two­factor authentication using Authy
- notification to RobinHood admin of new user sign­up
- send invitations to new users to join platform via e­mail
- edit user details (name, contact info, etc; enable / disable two­factor authentication)
- optionally limit access to registration through invite codes
- Wallet API and user interface components
- view pending and settled balances / holdings for USD or Euro or RobinHood shares and Identity Management
- view recent pending and settled transactions (up to 48 hours in the past)
- e­mail notification of incoming and outgoing transactions
- transfer assets to other users (identified by e­mail address or blockchain address)
- optional two­factor authentication when sending assets to other users
- support for multiple private keys per user account
- optional analytics (Google Analytics, Mixpanel) integration*

#### Payment Integrations
- optional two­factor authentication to transfer funds using Authy
- Detailed KYC/AML single user reports are available at additional fee passed through from our identity verification provider (Currently USD 5 per report)
- Send/Receive payments to shareholder members in the following manners via the website**

> Bitcoin
> ACH
> Wire Transfer
> SEPA Direct Debit

---

It is important here to note that this iteration of Robin Hood Coop is built using open source tools we are developing. It is the first product built ontop of the 'fertile financial ground'. 

The underlying financial fertiliser is conceived of as a ['Market Network'](http://techcrunch.com/2015/06/27/from-social-to-market-networks/). This should and will be commonly owned and governed.

**Comment by ghost** [comment]
#### ... Part Four TL;DR

As a basline we have a product which can handle automated bank account checking for payments in and payments out... AND payments WITHIN.

The next two paragraphs are worth repeating...

> It is important here to note that this iteration of Robin Hood Coop is built using open source tools we are developing. It is the first product built ontop of the 'fertile financial ground'. 
> 
> The underlying financial fertiliser is conceived of as a ['Market Network'](http://techcrunch.com/2015/06/27/from-social-to-market-networks/). This should and will be commonly owned and governed.

---

Next steps... 

Next thing we are working on is a more stream lined version of Setting Up, Proposing, Auditing and Vetting projects.

This is where {future}my.enspiral comes in.

For a more full look at {future}my.enspiral [go here](https://docs.google.com/document/d/1vC1S12xvnw68SLCVLc6wo5yrxQzVyf9kAiYUrN7mEs8/edit).

**Comment by bhaugen** [comment]
In case we need it for the VF treatment of the RobinHood stories, this [slide deck by Bill McCarthy](https://drive.google.com/file/d/0BxlqOeaPnXHfRmE3RFFNdVJfRWs/view?usp=sharing) has a fairly dense section about analyzing monetary instruments in REA that I _think_ I can understand well enough to explain. But I bet the RH gang will be way ahead of me... :grinning:

**Comment by fosterlynn** [comment]
@dan-mi-sun I'm studying, will come back asap with some beginning thoughts or summarized version of the use case as it relates to VF, or something.... might be a day or two.

> Next thing we are working on is a more stream lined version of Setting Up, Proposing, Auditing and Vetting projects. This is where {future}my.enspiral comes in.

I was thinking to take a look at all economic flows.  And I was thinking the same for my.enspiral.  This statement indicates a smaller scope, but I do understand this is part of the original pain point in Enspiral.  Maybe I'll start broad, and we can narrow from there.  That would at least help me.... :blush:

**Comment by ahdinosaur** [comment]
@dan-mi-sun this is amazing, thanks for spending the energy to share this level of detail with us. i'm keen to join in digesting this epic odyssey into smaller stories. :cat:

**Comment by ghost** [comment]
No problem... Will have more to report by the weekend as we're having a milestoning discovery session with ThoughtWorks tomorrow (+ Hylo hopefully).

Should be able to define some of the next steps that we will be taking.

**Comment by fosterlynn** [comment]
@dan-mi-sun questions:

> Users can transfer assets to other users as described in section 2. (“Payment Integrations”) below.
> RobinHood and RobinHood customers are able to transfer assets to any other user at any time (optionally subject to limitations set by RobinHood)

What is the difference between "users" and "customers"?  .... and "members"?

> RobinHood is able to issue an unlimited number of series of assets representing different share classes in RobinHood Cooperative

This is because I am extremely unfamiliar with anything to do with standard investing, I'm sure....  So where do these assets come from?  What are the share classes and the implications?

> transfer assets to other users (identified by e­mail address or blockchain address)

So all the statements in the new software related to transferring of assets freely are a new thing, where the old thing is you could only transfer as of exit from the co-op? 

> RobinHood is able to make distributions to shareholders (i.e. dividend payments)

 Same for getting dividends... before only on exit?  Now issued periodically?

Thanks!

**Comment by fosterlynn** [comment]
@dan-mi-sun @ahdinosaur OK, I've gone through the epic odyssey and taken notes!  Wow!  I have to admit, I understand a lot less about the new product than the old way.... my problem, just no experience with investing.

I think I'd like to start creating an abbreviated mapping of the epic odyssey to the VF vocabulary, a lot of which is under development.... so this will be a chicken-egg dialectical feedback process of some sort.... :)

I think I'll do it in parts, like Dan does....

**Comment by fosterlynn** [comment]
Starting with Agent vocabulary, initial thoughts, can get all of this into json-ld text and owl diagram at some point when there has been some discussion.  Reference the Agent repo.

Subclasses of Agent:
    - Cooperative (subclass of foaf:Group?)
    - BenefitCorporation (subclass of org:Organization, which we never agreed on using)
    - Person (foaf:Person)
    - Project (subclass of foaf:Group?)
    - SelectionCommittee
    - Board
[We already have Person.  But all of these should be standard in some way... maybe are in an existing vocab?  Project probably needs more discussion though, it is a can of worms.]

Agent relationship types (subproperties of rdf:Property)
    - vf:memberOf
    - rfc:angelFor (or adminFor, but "angel" is great!)
    - rfc:boardMemberOf (RFC - or just memberOf the Board?)
    - rfc:contributorTo
    - rfc:advisorTo
    - a few others, including "parasite"... good thing we let you define your own!  :) 

Agents:
    - Robin Hood Cooperative (a Cooperative)
    - Robin Hood Collective (a Group?)
    - Robin Hood Services (a BenefitCorporation)
    - RHC Selection Committee
    - RHC Board
    - each person is also an Agent

Relationships, a few examples to give the idea:
    - Teppo Vesikukka - contributorTo - Robin Hood Cooperative
    - Liisa Välikangas - memberOf - RHC Board
    - Dan Hassan - memberOf - Robin Hood Cooperative

UserAccount (we haven't defined this, but there are existing vocabs)

Membership states: Pending, Paid, Invested, Exited or something like that (we used to have a status on the relationship between Agents, probably need to bring it back?)

**Comment by ghost** [comment]
> What is the difference between "users" and "customers"? .... and "members"?

All these have been interchangable in the old form. This will continue to be the case within the walled gardens of RHC. 

> > RobinHood is able to issue an unlimited number of series of assets representing different share classes in RobinHood Cooperative
> 
> This is because I am extremely unfamiliar with anything to do with standard investing, I'm sure.... So where do these assets come from? What are the share classes and the implications?

These are created out of thin air. They should ideally be in a one to one relationship to funds invested by a member. These funds are then invested on the stock market via our brokerage account. I hope this answers your question?

> So all the statements in the new software related to transferring of assets freely are a new thing, where the old thing is you could only transfer as of exit from the co-op?

You could either get the coop to buy back shares, or you could email the coop asking for help in finding another member to buy the shares... this is now automated.

**Comment by ghost** [comment]
@fosterlynn let me know what I can do to help with this. I will add to the new way of doing things soon...

I'll make a screencast of the new thing within a couple of weeks should make things a bit easier to understand... for me at least! ;)

**Comment by gcassel** [comment]
I haven't had time to properly read and digest your epic sharings above @dan-mi-sun  but I'm really looking forward to it!    (Thank you)  :)

**Comment by fosterlynn** [comment]
Next installment: Resource.  We haven't formally done this vocab, so I'm using the REA/NRP vocabulary for now.  Let's fix it up for VF!

Economic Resource Types:
    - Bank account
    - Virtual account
    - Wallet?
    - Blockchain address
    - Investment share (maybe one for each share class? maybe a lot more specific, don't know?)
    - Investment asset????? (guessing)
    - Admin work
    - maybe some other kinds of work, even though not paid yet? (we use Economic Resource Types to define types of work, where you get as specific as you need for either different pay/distribution rates or categorization for reporting purposes; and there may be use in recording unpaid work just for understanding what it really takes to run the organization)

I'm sure we are missing some concrete things, like office space, etc.

Economic Resources:
    - RHCs Finnish Bank Account (ERT=Bank account)
    - RHCs Finnish 'holding account (ERT=Bank account?)
    - RHCs USA bank account (ERT=Bank account)
    - RHCs Interactive broker account (ERT=? do we need Broker account?)
    - a virtual account and/or Blockchain address for each person (member?) (ERT=Virtual account or Blockchain address; the virtual accounts would reference whatever bank account they are part of as an "underlying resource")
    - a virtual account for RCH, any other organizational entity that will record a chunk of money
    - probably some specific asset resources? (guessing)

More economic resources will be created during operations.... like when some kind of earmark for specific purposes is needed, say the holding account amounts on member exit, or for whatever future use; and maybe stuff around the investment results, I don't know how that part works....

Skipping recipes for now, don't know if that would be useful here.

OK, next up is the fun part, when it goes into motion....  :)

**Comment by fosterlynn** [comment]
> I'll make a screencast of the new thing within a couple of weeks should make things a bit easier to understand...

:+1:   Anyway, no worries, this will proceed at the pace it proceeds, and we'll fix what needs fixing.... :)

**Comment by fosterlynn** [comment]
Next installment: economic activity when a person joins RHC

Sharing scales:

> If a member decides to keep 100% of profits to themselves, then we have a one time tax of 10% of their total payment. This goes to Robin Hood Projects Fund. If a member chooses to share 50% of the profits with RHPF then they are not taxed on their payment.

I'm not sure the best way to record this choice.  It's a bit like Sensorica's value equations, but the work flow is different. Letting it simmer in background, possibly a discussion for later in the vocab development....

Making the following abbreviated because of the data structure discussions going on in other issues...

**Transfer**, type Membership fee, from agent Lynn, to agent Robin Hood Coop, quantity/unit €30, outflow resource RHC virtual account in RHC Finnish Bank Account (incremented)

**Exchange**, type Member purchases investment shares 
**Transfer**, type Payment, from agent Lynn, to agent Robin Hood Coop, quantity/unit €60, outflow resource RHC holding account (incremented)
**Transfer**, type Investment shares, from agent Robin Hood Coop, to agent Lynn, quantity/unit 2 shares, outflow resource Lynn's investment shares (incremented)

**Transfer**, type Investment tax, from agent Lynn, to agent Robin Hood Coop, quantity/unit €6, outflow resource RCH projects fund virtual account (incremented) 

[Is there a Lynn's virtual account involved here too?  As part of the brokerage account?  Not sure this is the way the shares work either, this assumes the out of thin air inflow.....?]

So, is this more or less seeming on the right track to people?

Drat, now is when I should be like @dan-mi-sun and offer a picture for reading this far....  too bad for y'all I'm not that creative....  :smile:

**Comment by ahdinosaur** [comment]
this is really great @fosterlynn. :)

as a reward for reading this far, here's a picture from [Patchwork](https://ssbc.github.io/patchwork) #art posted by @wanderer:

![cpu city](https://cloud.githubusercontent.com/assets/719605/14904500/a9f5fffa-0dfc-11e6-83ea-ca5a23499b9c.gif)

**Comment by fosterlynn** [comment]
@ahdinosaur hey great... can you do my reward for reading pictures from now on? :smile:

**Comment by ahdinosaur** [comment]
@fosterlynn for sure. :cat:

**Comment by ghost** [comment]
@bhaugen we are actually implementing a ['Conversation for Agreement'](https://github.com/valueflows/agent/issues/62#issuecomment-215613372) module within our latest integration work with Robin Hood / Hylo / Thought Works. I have been calling it the Offer/Bargain Engine.

Will have the user journies for these mapped out within the next week. Will post here when they are available.

**Comment by ghost** [comment]
Once we have rolled out the above features... one of the future implementations of 'groups' that we will be looking at will be something along the lines of this: https://opencollective.com/

Indeed, we may just work directly with them at OC to build out a node of thier network ontop of our protocols to see how that plays out.

**Comment by ghost** [comment]
[Robin Hood Recap.pdf](https://github.com/valueflows/exchange/files/241886/Robin.Hood.Recap.pdf)

Here are the user journies which came out of the RH / Hylo / TW integration roadmapping yesterday.

**Comment by fosterlynn** [comment]
A note: Right now I have reciprocal transfers held together by an exchange.  I think (tentatively) that a better way to model that is to create a many-to-many associative thing relating them.  (Sorry for data model speak.)  In general, to relate events (which are hidden in this issue so far because that structure is under discussion) to each other as a way to denote which are "because of" another event.  Simplest example of reciprocity: sell a share, pay for that share.  This would make it consistent with how we treat distributions, which are way too complicated to be held together by something like an exchange.  A topic for #16 or a new issue, but thought I should briefly mention it here too because of my examples.

**Comment by fosterlynn** [comment]
Next installment: Monthly activity

**Transfer**, type Admin salary, from agent Robin Hood Coop, to agent Fred, quantity/unit €1500, inflow resource RHC operations virtual account (decremented), outflow resource Fred's virtual account (incremented) 
-- or maybe just cut Fred a check? in which case there would be no outflow resource 
-- Fred could record his work every month too and get paid by the hour, but my impression is there is a salary?

**Transfer**, type Monthly investment, from agent Robin Hood Coop, to agent Robin Hood Coop, quantity/unit €20,000, inflow resource RFC holding account (decremented), outflow resource RFC Interactive broker account (incremented)

**Transfer**, type Operational funds, from agent Robin Hood Coop, to agent Robin Hood Coop, quantity/unit €2,000, inflow resource RFC holding account (decremented), outflow resource RFC operations virtual account (incremented)

**Transfer**, type Investment profits, from agent Wall Street investment firm or whatever), to agent Robin Hood Coop, quantity/unit €8,500, outflow resource RFC investment account (incremented)

**Transfer**, type Personal profit, from agent Robin Hood Coop, to agent Robin Hood Coop, quantity/unit €123, inflow resource RFC investment account (decremented), outflow resource Lynn's virtual account (incremented) (Lynn chose to pay 10% up front)
**Transfer**, type Personal profit, from agent Robin Hood Coop, to agent Robin Hood Coop, quantity/unit €456, inflow resource RFC investment account (decremented), outflow resource Mikey's virtual account (incremented)
**Transfer**, type RHC profits, from agent Robin Hood Coop, to agent Robin Hood Coop, quantity/unit €456, inflow resource RFC investment account (decremented), outflow resource RHC profits virtual account (incremented) (Mikey chose to give 50% of his profits)
**etc.** until all €8,500 personal profits are transferred

**Transfer**, type Operational funds, from agent Robin Hood Coop, to agent Robin Hood Coop, quantity/unit €500, inflow resource RFC profits virtual account (decremented), outflow resource RFC operations virtual account (incremented)

**Transfer**, type Project funding, from agent Robin Hood Coop, to agent Project 1, quantity/unit €800, inflow resource RFC profits virtual account (decremented), outflow resource Project 1 virtual account, part of US bank account (incremented)

**Transfer**, type Project payout, from agent Project 1, to agent Project 1, quantity/unit €800, inflow resource Project 1 virtual account (decremented)

Or something like this anyhow.  :)  OK, done for now, I'm sure this can be corrected where I goofed up, and fine tuned a whole bunch, when we want to get into such detail.....

Cue @ahdinosaur reward for reading this far.....

**Comment by ahdinosaur** [comment]
as a reward for reading this far (and especially to @dan-mi-sun and @fosterlynn who contribute such amazing work), here's another example of a fractal system [where everything is connected](https://github.com/valnet/valuenetwork/wiki/Everything-is-connected):

![infinite zooming fractal gears](http://i.imgur.com/B8eNodx.gif)

**Comment by ghost** [comment]
thanks @fosterlynn (& @ahdinosaur for the treats)

Here is another graphic I have just come across from [here](http://discourse.robinhoodcoop.org/t/moving-on-to-blockchain/259) (which I forgot we had done) which details our conception of how the new system would work (this before implementation... so a few things have changed)

[These user stories](https://github.com/valueflows/exchange/files/241886/Robin.Hood.Recap.pdf) relate to the path denoting the project establishment / funding and offering (which in the picture below are surrounded by red/pink dots).

We still need to wrap our head around how to harness the wisdom of the crowd in terms of vetting proposed projects & as to whether they 'expand the commons', or not...

![bc coop](https://cloud.githubusercontent.com/assets/6864509/14944741/6d1692ac-1040-11e6-948b-d1b5a00ab653.jpg)

**Comment by fosterlynn** [comment]
@dan-mi-sun nice!  Yeah, I didn't address the new system yet, and I think still will wait until you get all settled with your deployment and all.  And we will have questions. 

We talked today about focusing on Process for a bit because the vocabulary is more clear there and I think it will help us with our Transfer discussions.

If you get impatient for any reason, or want to take this in some direction, just ping! :)  We do want to keep up with live use cases as they unfold!

**Comment by ghost** [comment]
All good - we're (VF and RH) entangled now. 

Feedback on the first iteration would be very welcome - we're at the stage of formation, where input will help to effect each iteration of improvements in the system. 

i.e. you have in us an opportunity to encode some 'Value Flows' ways of doing things. From what I can tell we are already somewhat closely aligned.

**Comment by fosterlynn** [comment]
Entanglement :+1:

**Comment by fosterlynn** [comment]
@dan-mi-sun I'm trying to figure out how we should move forward from here, need help.  :)

A question: Are you thinking about the VF model being helpful internal to the app?  And/or are you thinking there will be situations where you actually want to communicate between apps using the VF vocab and linked open data?  Like either between pieces of your app or between your stuff and the outside world?

And another: what would be most helpful to RHC? I'm not sure about feedback on your first iteration, you already have identified a number of things to make better.  I could take a pass at your new stuff if that is helpful.

And a note, which @bhaugen reminded me of earlier:  It is always useful to move forward with as much as one can at the time, working in spirals.  So if VF can be embedded in RHC as much as we know now, it will then help VF mature as it gets used, help others towards interoperability, etc....and as RHC finds opportunities for improvement, rinse and repeat.

**Comment by coopchange** [comment]
woof - y'all brilliant minds are blowin me away every day.

> working in spirals
> +1

I know I have a bit of background work to do - @bhaugen has graciously
hooked me up with excellent reading material to get a fuller picture of
valueflo.ws; I have seen a presentation from RHC & saw some of @bhaugen's
presentation in Michigan that included their work, and I am scrambling
around the future-m.e channels (gdoc + slack + loomio), hopefully not too
noisily.

It seems that Enspiral is considering the f-m.e conversation as a R&D
project ATM, with team-building/convening being their top priority.

@dan-mi-sun has done some wonderful work laying out some roadmapping
through his gdoc where bhaugen, @ahdinosaur, others, & myself threw in our
intents to participate...

Craig Ambrose has gotten $1,000 in funding to create promotional materials
around "Paybuffer" - an app that is most directly used to stabilize
freelancer's income by automating good banking practices (ie putting a
certain amount into the bank for rainy days) => my impression is the goal
is that this works as an onboarding process for people to start
understanding the purpose of collective investment & empower the easy
formation of co-ops; definitely a piece (a.k.a. a front door/ storefront
window) of this evolving puzzle IMO.

@dan-mi-sun: I'm mostly posting as to start getting some more direction
with the three hours I have to put towards this project each week (and I
totes understand if noone has time to direct me in this regard! and would
be totally cool if folks thought I should wait on getting involved w/ my
coalition building, strategy & outreach skills) I'm pretty self-directed
and reading valueflo.ws will happen outside of the time-allotments (b.c. I
can't help myself).

peace n love, keep on keepin' on beautiful peeps :)

On Mon, May 2, 2016 at 9:10 AM, Lynn Foster notifications@github.com
wrote:

> @dan-mi-sun https://github.com/dan-mi-sun I'm trying to figure out how
> we should move forward from here, need help. :)
> 
> A question: Are you thinking about the VF model being helpful internal to
> the app? And/or are you thinking there will be situations where you
> actually want to communicate between apps using the VF vocab and linked
> open data? Like either between pieces of your app or between your stuff and
> the outside world?
> 
> And another: what would be most helpful to RHC? I'm not sure about
> feedback on your first iteration, you already have identified a number of
> things to make better. I could take a pass at your new stuff if that is
> helpful.
> 
> And a note, which @bhaugen https://github.com/bhaugen reminded me of
> earlier: It is always useful to move forward with as much as one can at the
> time, working in spirals. So if VF can be embedded in RHC as much as we
> know now, it will then help VF mature as it gets used, help others towards
> interoperability, etc....and as RHC finds opportunities for improvement,
> rinse and repeat.
> 
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly or view it on GitHub
> https://github.com/valueflows/exchange/issues/15#issuecomment-216230456

## 

Peace, Purpose, & Progress,
xander gieryn
5O3-430-45I3
_@coopchange_ http://twitter.com/coopchange
_pronouns: he/him/they/them_
Indy Art | Media Co-op
iAMCoop.org http://iamcoop.org/ | Tw http://twitter.com/iamcoop317 | Ig
http://instagram.com/iamcoop317
Indy Worker Justice Center
info@indywjc.org | Fb http://facebook.com/indyworkerjusticecenter
_~Dream 3 generations forward & Remember 3 generations back~_

**Comment by ghost** [comment]
> A question: Are you thinking about the VF model being helpful internal to the app? And/or are you thinking there will be situations where you actually want to communicate between apps using the VF vocab and linked open data? Like either between pieces of your app or between your stuff and the outside world?

At it's simplest this might look like: us shipping some code. Then going through the code with you guys and then changing names to comply with VF:naming, where the processes we have implemented LOOK and FEEL like a VF way of doing things - this should be easiest at this earliest stage as there are fewer people that push access has to get through. My thinking is that once the 'groundwork' (that is to say... lexicon which hoook and point through to wider VF processes and 'things') is laid it is much easier to go forward.

More complex is the actual modules themselves - do they comply with the principles of VF... this is something that will require getting deeper into the weeds.

> And another: what would be most helpful to RHC? I'm not sure about feedback on your first iteration, you already have identified a number of things to make better. I could take a pass at your new stuff if that is helpful.

Seeing your process of analysis is useful :)

That being said - we're investing in the new phase, so the most useful thing would be to analyse that when we ship. What looks like VF thinking? What is going in a direction which is closing down possibilities for VF approaches? etc etc.

> And a note, which @bhaugen reminded me of earlier: It is always useful to move forward with as much as one can at the time, working in spirals.

Agreed. I think my suggestions above draw from the Spiral thinking :)

**Comment by bhaugen** [comment]
@dan-mi-sun the VF model - the concepts and relationships, the ontology if you want to be fancy - is more important than the names.  The names only become important if you want to communicate with other agents on a shared vocabulary, cast your intents to the Web, or do something else that swims in the linked open data ocean. The model is what Bill McCarthy works on.

So if you get to the point of having something like a schema or data or object model or anything in that rough territory, that would be a good discussion topic. And we could help with that. (I think.)

**Comment by fosterlynn** [comment]
@dan-mi-sun @bhaugen yes seems like a great place to start would be doing some down and dirty mapping of the data and data relationships, in whatever form that appears in the code.  We will already have a sense of how it fits into the user stories.  As early as you have something makes sense.

**Comment by ghost** [comment]
@fosterlynn @bhaugen @ahdinosaur @gcassel @coopchange 

A real treat to come back to this years later! In the meantime there was the birth of a child and many continental migrations. Now that I have reconnected with this I will make some time to come back here and update on what happened during this span of time.

**Comment by fosterlynn** [comment]
Hey @dan-mi-sun great to see you back here!  And lovely to hear about your child, that'll put a crimp in your free time! In a good way, of course... :)

**Comment by almereyda** [comment]
We have moved the ValueFlows organization from GitHub to https://lab.allmende.io/valueflows.

This issue has been closed here, and all further discussion on this issue can be done at

> https://lab.allmende.io/valueflows/exchange/-/issues/15.

If you have not done so, you are very welcome to register at https://lab.allmende.io and join the ValueFlows organization there.
